src/main/java/de/hhu/stups/plues/ui/controller/Reports.java

```diff
@@ -185,67 +270,242 @@ private void displayModuleAbstractUnitUnitSemesterConflicts(final Store store,
+
+    private final Logger logger = Logger.getLogger(getClass().getSimpleName());
+    private final String faculty;
+    private final Map<String, String> resources;
+    private final List<Module> incompleteModules;
+    private final List<Module> impossibleModulesBecauseOfMissingElectiveAbstractUnits;
+    private final List<Course> impossibleCourses;
+    private final List<Course> impossibleCoursesBecauseOfImpossibleModules;
+    private final List<Course> impossibleCoursesBecauseOfImpossibleModuleCombinations;
+
+    RealReportData(final List<Module> incompleteModules,
+                   final List<Module> impossibleModulesBecauseOfMissingElectiveAbstractUnits,
+                   final List<Course> impossibleCourses,
+                   final List<Course> impossibleCoursesBecauseOfImpossibleModules,
+                   final List<Course> impossibleCoursesBecauseOfImpossibleModuleCombinations,
+                   final Map<Course, Set<Module>> mandatoryModules,
+                   final Map<Module, Set<AbstractUnit>> quasiMandatoryModuleAbstractUnits,
+                   final Set<Unit> redundantUnitGroups,
+                   final Map<Course, Map<Module, Set<AbstractUnit>>>
+                     impossibleCourseModuleAbstractUnits,
+                   final Map<Course, Map<Module, Set<Pair<AbstractUnit>>>>
+                     impossibleCourseModuleAbstractUnitPairs,
+                   final HashMap<Module,
+                     List<ModuleAbstractUnitUnitSemesterConflicts.Conflict>>
+                     moduleAbstractUnitUnitSemesterConflicts,
+                   final List<Unit> unitsWithoutAbstractUnits,
+                   final List<AbstractUnit> abstractUnitsWithoutUnits,
+                   final String faculty,
+                   final Map<String, String> resources) {
+      this.incompleteModules = incompleteModules;
+      this.impossibleModulesBecauseOfMissingElectiveAbstractUnits =
+        impossibleModulesBecauseOfMissingElectiveAbstractUnits;
+      this.impossibleCourses = impossibleCourses;
+      this.impossibleCoursesBecauseOfImpossibleModules =
+        impossibleCoursesBecauseOfImpossibleModules;
+      this.impossibleCoursesBecauseOfImpossibleModuleCombinations =
+        impossibleCoursesBecauseOfImpossibleModuleCombinations;
+      this.mandatoryModules = mandatoryModules;
+      this.quasiMandatoryModuleAbstractUnits = quasiMandatoryModuleAbstractUnits;
+      this.redundantUnitGroups = redundantUnitGroups;
+      this.impossibleCourseModuleAbstractUnits = impossibleCourseModuleAbstractUnits;
+      this.impossibleCourseModuleAbstractUnitPairs = impossibleCourseModuleAbstractUnitPairs;
+      this.moduleAbstractUnitUnitSemesterConflicts = moduleAbstractUnitUnitSemesterConflicts;
+      this.unitsWithoutAbstractUnits = unitsWithoutAbstractUnits;
+      this.abstractUnitsWithoutUnits = abstractUnitsWithoutUnits;
+
+      this.faculty = faculty;
+      this.resources = resources;
+    }
+
+    void print() {
```

Comment:
This should go in a PDFReport class, that only takes care of Rendering the Document based on a given RealReportData instance